File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/profile/GetPublicProfile.java
package com.moovie.backend.domain.usecase.profile;

import com.moovie.backend.domain.model.PublicProfile;
import com.moovie.backend.domain.repository.ProfileRepository;
import org.springframework.stereotype.Service;

@Service
public class GetPublicProfile {

    private final ProfileRepository profileRepository;

    public GetPublicProfile(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }

    public PublicProfile invoke(String userId) {
        return profileRepository.getPublicProfile(userId);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/profile/GetPublicProfile.kt
package org.mobyle.domain.usecase.profile

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.PublicProfile
import org.mobyle.domain.repository.ProfileRepository

class GetPublicProfile(private val repository: ProfileRepository) {
    operator fun invoke(userId: String): PublicProfile = runBlocking {
        repository.getPublicProfile(userId)
    }
}
